sdk/tests/wallet/balance.rs

```diff
@@ -7,11 +7,115 @@ use iota_sdk::{
         unlock_condition::{AddressUnlockCondition, ExpirationUnlockCondition},
         BasicOutputBuilder, UnlockCondition,
     },
-    wallet::Result,
+    wallet::{
+        account::types::{AccountBalance, AccountBalanceDto},
+        Result,
+    },
 };
 
 use crate::wallet::common::{create_accounts_with_funds, make_wallet, setup, tear_down};
 
+#[test]
+fn balance_to_dto() {
+    let balance = AccountBalance::rand_mock();
+    let balance_dto = AccountBalanceDto::from(&balance);
+
+    assert_eq!(balance.base_coin().total().to_string(), balance_dto.base_coin.total);
+    assert_eq!(
+        balance.base_coin().available().to_string(),
+        balance_dto.base_coin.available
+    );
+    #[cfg(feature = "participation")]
+    assert_eq!(
+        balance.base_coin().voting_power().to_string(),
+        balance_dto.base_coin.voting_power
+    );
+
+    assert_eq!(
+        balance.required_storage_deposit().alias().to_string(),
+        balance_dto.required_storage_deposit.alias
+    );
+    assert_eq!(
+        balance.required_storage_deposit().basic().to_string(),
+        balance_dto.required_storage_deposit.basic
+    );
+    assert_eq!(
+        balance.required_storage_deposit().foundry().to_string(),
+        balance_dto.required_storage_deposit.foundry
+    );
+    assert_eq!(
+        balance.required_storage_deposit().nft().to_string(),
+        balance_dto.required_storage_deposit.nft
+    );
+
+    assert_eq!(balance.native_tokens().len(), balance_dto.native_tokens.len());
+    assert_eq!(balance.nfts().len(), balance_dto.nfts.len());
+    assert_eq!(balance.aliases().len(), balance_dto.aliases.len());
+    assert_eq!(balance.foundries().len(), balance_dto.foundries.len());
+    assert_eq!(
+        balance.potentially_locked_outputs().len(),
+        balance_dto.potentially_locked_outputs.len()
+    );
```

Comment:
I don't think these tests are doing much. You should just do a roundtrip and it would test everything at once.

Reply:
What API should I use to do the roundtrip?